test/src/test/java/org/corfudb/integration/LogReplicationSinkWriterIT.java

```diff
@@ -0,0 +1,289 @@
+        // Also verify all the table's contents for snapshot sync
+        verifyTableContentsAndRecords(1, 2, 4);
+
+
+        /* The last 4 tables are for log entry sync */
+
+        // Stop LR to modify Sink side registry table in advance
+        stopActiveLogReplicator();
+        // Clear these two map as different set of tables will be verified
+        mapNameToMapActive.clear();
+        mapNameToMapStandby.clear();
+
+        // Table005 and Table006 - Open in both Source and Sink, while on Sink side they will be opened with
+        // different metadata. Verify that Sink side's records are retained.
+        // Table007 and Table008 - Open only in Source side. Verify that their records are replicated to Sink.
+        prepareTablesToVerify(5, 6, 8);
+
+        // Restart Source LR after opening the tables
+        startActiveLogReplicator();
+
+        // Write data to Source side, note that mapNameToMapActive now only have the last 4 tables
+        writeToActive(0, NUM_WRITES);
+
+        // At this point mapNameToMapStandby only has 2 tables. This step is also used as a barrier to indicate
+        // it's a good point to check records on Sink side
+        verifyDataOnStandby(NUM_WRITES);
+        // Verify Sink side records for Table005 and Table006 are retained, and records for Table007
+        // and Table008 are replicated.
+        // Also verify all the table's contents for snapshot sync
+        verifyTableContentsAndRecords(5, 6, 8);
+    }
+
+    /**
+     * Open tables on Source and Sink side accordingly.
+     *
+     * Tables from start to splitter - Open on both Source and Sink side, while on Sink side they will be
+     * opened with different metadata.
+     *
+     * Tables from splitter to end - Open on Source side only, after log replication verify their records
+     * exist in Sink side registry table.
+     */
+    private void prepareTablesToVerify(int start, int splitter, int end) throws Exception {
+        // From start to splitter - Open in both Source and Sink, while on Sink side they will be opened with
+        // different metadata. Verify that Sink side's records are retained.
+        for (int i = start; i <= splitter; i++) {
+            String mapName = TABLE_PREFIX + i;
+            Table<StringKey, IntValueTag, Metadata> tableActive =
+                    corfuStoreActive.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(IntValueTag.class, TableOptions.builder().build()));
+            mapNameToMapActive.put(mapName, tableActive);
+
+            Table<StringKey, IntValueTag, Metadata> tableStandby =
+                    corfuStoreStandby.openTable(NAMESPACE, mapName, StringKey.class,
+                            IntValueTag.class, Metadata.class,
+                            TableOptions.fromProtoSchema(NotFederatedTag.class, TableOptions.builder().build()));
```

Comment:
I creating a new Protobuf msg for making the code here self-explanatory is worth it, and I didn't see the harm for creating it in our sample Protobuf suite. If we just pick another one like "IntValue", the name is a bit confusing and we are using the default `false` value instead of explicitly set them

Reply:
We should not create a new protobuf if one already exists that fulfills the need.  Please reuse the existing code.

The name is also self-explanatory "IntValueTag" vs "IntValue".  We also have comments to explain what the test does and reading them is absolutely necessary to understand the tests.  There is no real value in adding a new redundant protobuf